src/frontend/src/helpers/Compiler/OnlineCompiler.ts

```diff
@@ -100,4 +103,12 @@ class OnlineCompiler extends AbstractCompiler {
       pid: this.activePIDs[0]
     });
   }
+
+  protected programDone(pid: number) {
```

Comment:
You don't call programDone when you're running tests, nor does programDone work when tests are being run.  Maybe I've overlooked something, but where do we clear the list of running PIDs when we're running tests?